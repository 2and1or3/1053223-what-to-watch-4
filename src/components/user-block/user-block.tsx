import React from "react";
import PropTypes from "prop-types";
import {Link} from "react-router-dom";
import {connect} from "react-redux";

import {UserStatus, AppRoute} from '../../consts.js';
import {getUserStatus} from '../../reducer/user/selectors.js';

const UserBlock = (props) => {
  const {authStatus} = props;

  const userElement = authStatus === UserStatus.AUTH ?

    (<div className="user-block__avatar">
      <Link to = {AppRoute.LIST}>
        <img src="img/avatar.jpg" alt="User avatar" width="63" height="63" />
      </Link>
    </div>) :
    (<Link
      className="user-block__sign-in"
      to = {AppRoute.LOGIN}
      style = {{color: `#dfcf77`, textDecoration: `none`}}>
      Sign In
    </Link>);

  return (
    <div className="user-block">
      {userElement}
    </div>
  );
};

UserBlock.propTypes = {
  authStatus: PropTypes.string.isRequired,
};

const mapStateToProps = (state) => ({
  authStatus: getUserStatus(state),
});

const ConnectedUserBlock = connect(mapStateToProps, null)(UserBlock);

export {UserBlock};
export default ConnectedUserBlock;
