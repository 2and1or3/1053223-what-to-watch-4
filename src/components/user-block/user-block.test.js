import React from "react";
import renderer from "react-test-renderer";
import {BrowserRouter} from "react-router-dom";

import {UserBlock} from './user-block.jsx';


describe(`Render UserBlock component`, () => {
  it(`Without auth`, () => {
    const tree = renderer
    .create(
        <BrowserRouter>
          <UserBlock authStatus = {`NO_AUTH`}/>
        </BrowserRouter>
    )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With auth`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <UserBlock authStatus = {`AUTH`}/>
          </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

// npm run test.jest -- components/user-block/user-block.test.js
