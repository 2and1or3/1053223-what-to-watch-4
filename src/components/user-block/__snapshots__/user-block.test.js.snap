// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render UserBlock component With auth 1`] = `
<div
  className="user-block"
>
  <div
    className="user-block__avatar"
  >
    <a
      href="/mylist"
      onClick={[Function]}
    >
      <img
        alt="User avatar"
        height="63"
        src="img/avatar.jpg"
        width="63"
      />
    </a>
  </div>
</div>
`;

exports[`Render UserBlock component Without auth 1`] = `
<div
  className="user-block"
>
  <a
    className="user-block__sign-in"
    href="/login"
    onClick={[Function]}
    style={
      Object {
        "color": "#dfcf77",
        "textDecoration": "none",
      }
    }
  >
    Sign In
  </a>
</div>
`;
